cmd/controller-manager/main.go

```diff
@@ -63,34 +68,47 @@ func main() {
-	flag.StringVar(&itsName, "its-name", "", "name of the Inventory and Transport Space to connect to (empty string means to use the only one)")
-	flag.StringVar(&wdsName, "wds-name", "", "name of the workload description space to connect to")
-	flag.StringVar(&allowedGroupsString, "api-groups", "", "list of allowed api groups, comma separated. Empty string means all API groups are allowed")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+	var controllers []string
+	pflag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The [host]:port from which /metrics is served.")
+	pflag.StringVar(&pprofAddr, "pprof-bind-address", ":8082", "The [host]:port fron which /debug/pprof is served.")
+	pflag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	pflag.StringVar(&itsName, "its-name", "", "name of the Inventory and Transport Space to connect to (empty string means to use the only one)")
+	pflag.StringVar(&wdsName, "wds-name", "", "name of the workload description space to connect to")
+	pflag.StringVar(&allowedGroupsString, "api-groups", "", "list of allowed api groups, comma separated. Empty string means all API groups are allowed")
+	pflag.StringSliceVar(&controllers, "controllers", []string{}, "list of controllers to be started by the controller manager, lower case and comma separated, e.g. 'binding,status'. If not specified (or emtpy list specifed), all controllers are started. Currently available controllers are 'binding' and 'status'.")
+	pflag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
 
-	itsClientLimits := clientopts.NewClientLimits[*flag.FlagSet]("its", "accessing the ITS")
-	wdsClientLimits := clientopts.NewClientLimits[*flag.FlagSet]("wds", "accessing the WDS")
-	itsClientLimits.AddFlags(flag.CommandLine)
-	wdsClientLimits.AddFlags(flag.CommandLine)
+	itsClientLimits := clientopts.NewClientLimits[*pflag.FlagSet]("its", "accessing the ITS")
+	wdsClientLimits := clientopts.NewClientLimits[*pflag.FlagSet]("wds", "accessing the WDS")
+	itsClientLimits.AddFlags(pflag.CommandLine)
+	wdsClientLimits.AddFlags(pflag.CommandLine)
 	klog.InitFlags(nil)
-	flag.Parse()
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
 	ctx := context.Background()
 	logger := klog.FromContext(ctx)
 	ctrl.SetLogger(logger)
 	setupLog := ctrl.Log.WithName("setup")
 
-	flag.VisitAll(func(flg *flag.Flag) {
+	pflag.VisitAll(func(flg *pflag.Flag) {
 		setupLog.Info("Command line flag", "name", flg.Name, "value", flg.Value)
 	})
 
 	// parse allowed resources string
 	allowedGroupsSet := util.ParseAPIGroupsString(allowedGroupsString)
 
+	// check controllers flag
+	ctlrsToStart := sets.New(controllers...)
+	if !sets.New(
+		strings.ToLower(binding.ControllerName),
+		strings.ToLower(status.ControllerName),
+	).IsSuperset(ctlrsToStart) {
```

Comment:
(1) You could use sets.Difference to discover the invalid values, if any, and then include them in the error message.

(2) Why bother changing the case? Why not let the string values in `binding.ControllerName` and `status.ControllerName` be the very same string values used in the `controllers` array?

Neither of these is critical.